lib/ffmpeg/transcoder.rb

```diff
@@ -56,8 +72,59 @@ def encoded
     end
 
     private
+    def pre_encode_if_necessary
+      # Don't pre-encode single inputs since it doesn't need any size conversion
+      return if @movie.interim_paths.size <= 1
+
+      # Set a minimum frame rate
+      output_frame_rate = [@raw_options[:frame_rate] || @movie.frame_rate, 30].max
+      output_frame_rate = 30 if output_frame_rate > 300
+
+      # Add a subset of the full encode options
+      pre_encode_options = @raw_options.is_a?(EncodingOptions) ? @raw_options.to_s_minimal : @raw_options
+
+      # Convert the individual videos into a common format, using the first video in as the "resolution"
+      @movie.paths.each_with_index do |path, index|
+        command = "#{@movie.ffmpeg_command} -y -i #{path} -movflags faststart #{pre_encode_options} -r #{output_frame_rate} -filter_complex \"[0:v]scale=#{@movie.height}:#{@movie.width},setsar=1[Scaled]\" -map \"[Scaled]\" -map \"0:a\" #{@movie.interim_paths[index]}"
+
+        FFMPEG.logger.info("Running transcoding...\n#{command}\n")
+        output = ""
+
+        Open3.popen3(command) do |stdin, stdout, stderr, wait_thr|
+          begin
+            yield(0.0) if block_given?
+            next_line = Proc.new do |line|
+              fix_encoding(line)
+              output << line
+              # if line.include?("time=")
+              #   if line =~ /time=(\d+):(\d+):(\d+.\d+)/ # ffmpeg 0.8 and above style
+              #     time = ($1.to_i * 3600) + ($2.to_i * 60) + $3.to_f
+              #   else # better make sure it wont blow up in case of unexpected output
+              #     time = 0.0
+              #   end
+              #   progress = time / @movie.duration
+              #   yield(progress) if block_given?
+              # end
```

Comment:
delete commented out code

Reply:
Added a `TODO` since this is something that should be implemented in the future.